Tests/AsyncHTTPClientTests/HTTPClientInternalTests.swift

```diff
@@ -217,7 +230,9 @@ class HTTPClientInternalTests: XCTestCase {
             func didFinishRequest(task: HTTPClient.Task<Response>) throws {}
         }
 
-        let httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+        // cannot test with NIOTS as `maxMessagesPerRead` is not supported
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
cc @weissi 
@adam-fowler if we create ELG here we need to make sure to stop it

Reply:
done